app/jobs/sync_with_blue_alliance_job.rb

```diff
@@ -11,40 +11,35 @@ def event_list(year = Time.now.year.to_s)
 		end
 	end
 
-	def team_list()
-		page = 0
-		teams = []
-		response = self.class.get("/teams/#{page}")
-		while response.success? && !response.parsed_response.empty?
-			teams += response.parsed_response
-			page += 1
-			response = self.class.get("/teams/#{page}")
-		end
-		teams
+	def team_list(event_key)
+		response = self.class.get("/event/#{event_key}/teams/simple")
+		response.parsed_response
 	end
 	
 	def perform()
 		old_logger = ActiveRecord::Base.logger
 		ActiveRecord::Base.logger = nil
 		Team.delete_all
 		Competition.delete_all
+		puts "Downloading events list..."
 		events = event_list()
+		puts "Adding events to database..."
 		events_done = 0
 		events.each do |tba_competition|
-			competition = Competition.new(:name => tba_competition['name'], :location => tba_competition['location_name'], :start_date => tba_competition['start_date'], :end_date => tba_competition['end_date'])
+			teams = team_list(tba_competition['key'])
+			teams2 = []
+			teams.each do |tba_team|
+				if !Team.exists?(number: tba_team['team_number'])
```

Comment:
It would be easier to use `team = Team.find_or_create_by(number: tba_team['team_number'], name: tba_team['nickname'])`. This will create any missing records and return a team object. If the team exists it will simply return the team.